chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt

```diff
@@ -593,16 +619,50 @@ object RoomSpek : Spek({
         }
 
         it("joins room") {
-            setUpInstanceWith(createDefaultRole(), newUsers(PUSHERINO), newRoom(GENERAL))
+            setUpInstanceWith(
+                    createDefaultRole(),
+                    newUsers(PUSHERINO),
+                    newRoom(GENERAL))
 
-            val pusherino = chatFor(SUPER_USER).connect().assumeSuccess()
+            val pusherino = chatFor(PUSHERINO).connect().assumeSuccess()
+            val superUser = chatFor(SUPER_USER).connect().assumeSuccess()
 
-            val room = pusherino.joinRoom(pusherino.generalRoom)
+            val generalRoomId = superUser.rooms[0].id
+            val room = pusherino.joinRoom(generalRoomId)
```

Comment:
I don't understand the motivation for the change in this test as the previous version was simpler and did not have to involve super user. Indeed with the proposed [patch](https://github.com/pusher/chatkit-android/pull/154/files#r372509094) it can get even simpler :)

Reply:
I explained [here](https://github.com/pusher/chatkit-android/pull/154#discussion_r372343810) why i had to get the super user involved - the previous test wasn't actually testing the user joining a room they previously weren't a member of.